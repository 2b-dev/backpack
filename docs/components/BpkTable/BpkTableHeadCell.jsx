import React from 'react'
import CssModules from 'react-css-modules'

import styles from './bpk-table.scss'

const BpkTableHeadCell = (props) => <th styleName='bpk-table__cell bpk-table__cell--head' {...props} />

export default CssModules(BpkTableHeadCell, styles, { allowMultiple: true })
