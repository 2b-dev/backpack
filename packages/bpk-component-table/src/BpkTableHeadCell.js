import PropTypes from 'prop-types';
import React from 'react';
import { cssModules } from 'bpk-react-utils';

import STYLES from './bpk-table.scss';

const getClassName = cssModules(STYLES);

const BpkTableHeadCell = props =>
  <th
    className={['bpk-table__cell', 'bpk-table__cell--head'].map(getClassName).join(' ')}
  >
    {props.children}
  </th>;

BpkTableHeadCell.propTypes = {
  children: PropTypes.node.isRequired,
};

export default BpkTableHeadCell;
