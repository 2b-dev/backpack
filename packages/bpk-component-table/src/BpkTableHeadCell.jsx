import React, { PropTypes } from 'react'

import './bpk-table.scss'

const BpkTableHeadCell = ({ children }) => <th className='bpk-table__cell bpk-table__cell--head' children={children} />

BpkTableHeadCell.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
}

export default BpkTableHeadCell
