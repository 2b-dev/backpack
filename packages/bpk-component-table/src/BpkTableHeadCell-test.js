jest.unmock('./BpkTableHeadCell.jsx')

import React from 'react'
import { shallow } from 'enzyme'
import BpkTableHeadCell from './BpkTableHeadCell.jsx'

describe('BpkTableHeadCell', () => {
  it('should render the correct element', () => {
    const wrapper = shallow(<BpkTableHeadCell>Heading</BpkTableHeadCell>)
    expect(wrapper.text()).toEqual('Heading')
    expect(wrapper.name()).toEqual('th')
  })

  it('should render the correct class', () => {
    const wrapper = shallow(<BpkTableHeadCell>Heading</BpkTableHeadCell>)
    expect(wrapper.prop('className')).toEqual('bpk-table__cell bpk-table__cell--head')
  })
})
