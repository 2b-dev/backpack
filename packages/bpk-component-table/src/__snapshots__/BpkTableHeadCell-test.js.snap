exports[`BpkTableHeadCell should render correctly 1`] = `
<th
  className="bpk-table__cell bpk-table__cell--head">
  Heading
</th>
`;
